lib/modifiers/batch.js

```diff
@@ -0,0 +1,171 @@
+var through = require('through2');
+var _ = require('lodash');
+
+/**
+ * Returns a new stream that batches writes based on a specific interval, count of items, or both.
+ * All items read on the stream and will be re-emitted as part of an array of one or more items
+ * based on the criteria defined in the options.
+ *
+ * @method
+ * @static
+ * @since    1.5.0
+ * @category Util
+ *
+ * @param   {Object} [options]       - Define how items will be batched and written to the
+ *                                     the stream. Note that when both `time` and `count` are
+ *                                     defined, items will be written to the stream whenever
+ *                                     either one of the conditions is met.
+ * @param   {Number} [options.time]  - The interval, in milliseconds, to limit writes to.
+ *                                     During that time, all items read on the stream will
+ *                                     be collected and written as an array at the set
+ *                                     interval.
+ * @param   {Number} [options.count] - The number of items to buffer before writing all
+ *                                     of them in an array.
+ * @returns {Stream}                 - Transform stream.
+ *
+ * @example
+ * var input = through.obj();
+ *
+ * input.pipe(nodeStream.batch({ time: 100 }));
+ *
+ * input.write(1);
+ *
+ * setTimeout(function() {
+ *   input.write(2);
+ *   input.write(3);
+ * }, 100);
+ *
+ * setTimeout(function() {
+ *   input.write(4);
+ *   input.write(5);
+ * }, 200);
+ *
+ * // => [[1], [2, 3], [4, 5]]
+ * @example
+ * var input = through.obj();
+ *
+ * input.pipe(nodeStream.batch({ count: 2 }));
+ *
+ * input.write(1);
+ * input.write(2);
+ * input.write(3);
+ * input.write(4);
+ * input.write(5);
+ *
+ * // => [[1, 2], [3, 4], [5]]
+ */
+function batch(options) {
+  var settings = _.extend({
```

Comment:
Yeah, I started down the path of emitting errors on the stream since they'll be caught by proper error handling and logged or in some other way handled. As long as stream errors are handled, there's no worry of the app crashing, and if they aren't handled, the app will crash. So, theoretically, it allows an implementer to get to a state where their app won't crash as long as they handle errors, and fix it at a later time since these aren't totally critical errors.

It's probably not the best argument, but it was my thought process for `pluck` and others.